src/mappings/shared/handleTokenEntity.ts

```diff
@@ -1,44 +1,166 @@
-import { create, getOptional } from '@kodadot1/metasquid/entity'
+import { create, getOptional, getWith } from '@kodadot1/metasquid/entity'
 import md5 from 'md5'
 import { CollectionEntity as CE, NFTEntity as NE, TokenEntity as TE } from '../../model'
-import { warn } from '../utils/logger'
+import { debug, warn } from '../utils/logger'
 import { Context } from '../utils/types'
 
 const OPERATION = 'TokenEntity' as any
 
-export async function handleTokenEntity(context: Context, collection: CE, nft: NE): Promise<TE | undefined> {
-  const nftMedia = nft.image || nft.media
+function generateTokenId(collectionId: string, nftMedia: string): string {
+  return `${collectionId}-${md5(nftMedia)}`
+}
+
```

Comment:
move to `src/mappings/shared/tokens/utils.ts`

Reply:
yes sir